### PR TITLE
Account for scale in scroll logic for `Parallax2D`

### PR DESCRIPTION
**Pull request by markdibarry**
Previously, it was misunderstood how the scaling should be considered when scrolling. I originally just kinda copied how `ParallaxLayer` handles it, and recently discovered that it handles it incorrectly. Scaled `ParallaxLayer`s currently wrap at incorrect points. This can be fixed in a separate PR.

This change is for `Parallax2D` to have its scale not taken into account when the repeat is spaced, but the point in which it wraps when scrolling is updated. It shouldn't come up very often as it's not a recommended workflow, but it's good to handle it either way. I also should add some notes to the docs in the future for "use at your own risk" scenarios like scaling or moving parents.

### CODE REVIEW

File: scene/2d/parallax_2d.cpp [hunk 1]
```diff
@@ -123,13 +123,10 @@ void Parallax2D::_update_scroll() {
 }
 
 void Parallax2D::_update_repeat() {
-	if (!is_inside_tree()) {
-		return;
```
Comment (KoBeWi) [inline scene/2d/parallax_2d.cpp:127]: Why did you remove this?

Comment (markdibarry) [reply scene/2d/parallax_2d.cpp:127]: Someone commented on another one of my PRs that early returns should only be done if the logic after is more than a couple lines or can branch in different ways. I think it was for `_validate_property()` and `_camera_moved()`. Since the change shortens it to only two lines now, I imagined someone would request it to be inside the if. It's not important to me, so whichever you prefer.

Comment (KoBeWi) [reply scene/2d/parallax_2d.cpp:127]: Well it was already there, so it's fine to leave it.

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!
